Sources/Sentry/SentryNSProcessInfoWrapper.mm

```diff
@@ -0,0 +1,50 @@
+#import "SentryNSProcessInfoWrapper.h"
+#import "SentryDependencyContainer.h"
+#import "SentryNSNotificationCenterWrapper.h"
+
+@implementation SentryNSProcessInfoWrapper
+
+- (NSProcessInfoThermalState)thermalState
+{
+    return NSProcessInfo.processInfo.thermalState;
+}
+
+- (BOOL)isLowPowerModeEnabled
+{
+    return NSProcessInfo.processInfo.isLowPowerModeEnabled;
+}
+
+- (void)monitorForPowerStateChanges:(id)target callback:(SEL)callback
+{
+    // According to Apple docs: "This notification is posted on the global dispatch queue. The
+    // object associated with the notification is NSProcessInfo.processInfo."
+    [SentryDependencyContainer.sharedInstance.notificationCenterWrapper
+        addObserver:target
+           selector:callback
+               name:NSProcessInfoPowerStateDidChangeNotification
+             object:NSProcessInfo.processInfo];
```

Comment:
`m`: Why do we need to specify `NSProcessInfo.processInfo` here? According to the [docs](https://developer.apple.com/documentation/foundation/nsnotificationcenter/1415360-addobserver) 

> The object that sends notifications to the observer. Specify a notification sender to deliver only notifications from this sender.

So the sender would be `NSProcessInfo.processInfo`. I don't think we need that. Passing `nil` should be fine.

Reply:
Annoyingly, this tidbit is only in a comment in NSProcessInfo.h, but it's not a headerdoc comment and doesn't appear in their online docs or in the xcode docs library:
> The object associated with the notification is NSProcessInfo.processInfo.
However I didn't test it with `nil` instead, I can try that.